### ISSUE TITLE
ITs fail

### ISSUE THREAD

**Issue by phax** [mojohaus/jaxws-maven-plugin#18]
The integration test for https://java.net/jira/browse/JAX_WS_COMMONS-98 does not throw an NPE, but it makes the Travis build fail.
Please configure the IT to handle this failure

**Comment by andham** [comment]
This is the behavior I see with the builds we have at Travis:

**OpenJDK 7:**
[ERROR] The following builds failed:
[ERROR] *  jaxwscommons-43/pom.xml
[ERROR] *  jaxwscommons-81/pom.xml

**Oracle JDK 7:**
[ERROR] The following builds failed:
[ERROR] *  jaxwscommons-43/pom.xml

**Oracle JDK 8:**
[ERROR] The following builds failed:
[ERROR] *  jaxwscommons-4/pom.xml
[ERROR] *  jaxwscommons-43/pom.xml
[ERROR] *  jaxwscommons-49/pom.xml
[ERROR] *  jaxwscommons-62/pom.xml
[ERROR] *  jaxwscommons-87/pom.xml
[ERROR] *  jaxwscommons-95/pom.xml

However, when building with Maven 3.3.3 and Oracle JDK 1.7.0_79 on MacOS 10.12 all ITs pass.

Looking into jaxwscommons-43 the specified mavenOpts for the test (-Dmaven.test.skip=true) isn't honored. This caused the IT to report an error.

**Comment by phax** [comment]
Maybe this should be changed to `-DskipTests=true` or just `-DskipTests` in order to work.

Output of `mvn --show-version --errors --batch-mode -Prun-its clean verify -Dittest.debug=false -Dinvoker.streamLogs=true` on Windows 10, Maven 3.3.9 and JDK 1.8.0_112:

> [ERROR] The following builds failed:
> [ERROR] *  jaxwscommons-4\pom.xml
> [ERROR] *  jaxwscommons-49\pom.xml
> [ERROR] *  jaxwscommons-62\pom.xml
> [ERROR] *  jaxwscommons-87\pom.xml
> [ERROR] *  jaxwscommons-95\pom.xml

**Comment by andham** [comment]
This seems to be an issue related to Linux. Tony Chemit has verified that build fails for him locally on Linux:

> Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
> Maven home: /opt/maven
> Java version: 1.8.0_121, vendor: Oracle Corporation
> Java home: /usr/lib/jvm/java-8.0_121-oracle/jre
> Default locale: fr_FR, platform encoding: UTF-8
> OS name: "linux", version: "4.4.0-63-generic", arch: "amd64", family: "unix"

> [ERROR] The following builds failed:
> [ERROR] *  jaxwscommons-87/pom.xml
> [ERROR] *  jaxwscommons-95/pom.xml
> [ERROR] *  jaxwscommons-62/pom.xml
> [ERROR] *  jaxwscommons-4/pom.xml
> [ERROR] *  jaxwscommons-49/pom.xml

**Comment by phax** [comment]
I'm running my tests with an additional `-Djavax.xml.accessExternalSchema=all` on the cmdline now

**Comment by andham** [comment]
OK, I see the same issues as @phax on Windows 7 with Maven 3.3.9 and Oracle JDK 1.8.0_112. However, if I use Oracle JDK 1.7.0_79 all ITs pass. WTF!

**Comment by andham** [comment]
The issues on Windows (and JDK 8) was due to #4. I've updated master to use JAX-WS RI 2.2.10 and that should now fixed that.
However, we still see build errors on Travis CI. jaxwscommons-43 mainly, but also jaxwscommons-81 on openjdk7.

**Comment by phax** [comment]
Here are my errors on jaxwscommons-4 test - seems to be an issue with the external schema access:
```
[INFO] Building: jaxwscommons-4\pom.xml
[INFO] [INFO] Error stacktraces are turned on.
[INFO] [INFO] Scanning for projects...
[INFO] [INFO]                                                                         
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] Building JAX_WS_COMMONS-4 2.5-SNAPSHOT
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] 
[INFO] [INFO] --- jaxws-maven-plugin:2.5-SNAPSHOT:wsimport (id1) @ jaxwscommons-4 ---
[INFO] [INFO] Processing: jar:file:/P:/git/jaxws-maven-plugin/target/it-repo/org/jvnet/jax-ws-commons/tests/wsdl-lib/1.0-SNAPSHOT/wsdl-lib-1.0-SNAPSHOT.jar!/ProcessOrder.wsdl
[INFO] [INFO] jaxws:wsimport args: [-keep, -s, 'P:\git\jaxws-maven-plugin\target\it tests\jaxwscommons-4\target\generated-sources\wsimport', -d, 'P:\git\jaxws-maven-plugin\target\it tests\jaxwscommons-4\target\classes', -verbose, -encoding, UTF-8, -Xnocompile, -p, sample, "jar:file:/P:/git/jaxws-maven-plugin/target/it-repo/org/jvnet/jax-ws-commons/tests/wsdl-lib/1.0-SNAPSHOT/wsdl-lib-1.0-SNAPSHOT.jar!/ProcessOrder.wsdl"]
[INFO] WSDL wird geparst ...
[INFO] 
[INFO] 
[INFO] [WARNING] schema_reference: Schemadokument 'ProcessOrder.xsd' konnte nicht gelesen werden, weil der 'file'-Zugriff wegen der von der Eigenschaft accessExternalSchema festgelegten Einschr寫ung nicht zul崳ig ist. 
[INFO]   Zeile 28 von jar:file:/P:/git/jaxws-maven-plugin/target/it-repo/org/jvnet/jax-ws-commons/tests/wsdl-lib/1.0-SNAPSHOT/wsdl-lib-1.0-SNAPSHOT.jar!/ProcessOrder.wsdl#types?schema1
[INFO] 
[INFO] Exception in thread "main" java.lang.reflect.InvocationTargetException
[INFO] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[INFO] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[INFO] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[INFO] 	at java.lang.reflect.Method.invoke(Method.java:498)
[INFO] 	at org.codehaus.mojo.jaxws.Invoker.main(Invoker.java:78)
[INFO] Caused by: java.lang.AssertionError: org.xml.sax.SAXParseException; systemId: jar:file:/P:/git/jaxws-maven-plugin/target/it-repo/com/sun/xml/bind/jaxb-xjc/2.2.8-b01/jaxb-xjc-2.2.8-b01.jar!/com/sun/tools/xjc/reader/xmlschema/bindinfo/binding.xsd; lineNumber: 52; columnNumber: 88; schema_reference: Schemadokument 'xjc.xsd' konnte nicht gelesen werden, weil der 'file'-Zugriff wegen der von der Eigenschaft accessExternalSchema festgelegten Einschr寫ung nicht zul崳ig ist. 
[INFO] 	at com.sun.tools.xjc.SchemaCache.newValidator(SchemaCache.java:80)
[INFO] 	at com.sun.tools.xjc.reader.internalizer.SCDBasedBindingSet.apply(SCDBasedBindingSet.java:237)
[INFO] 	at com.sun.tools.xjc.ModelLoader.createXSOM(ModelLoader.java:541)
[INFO] 	at com.sun.tools.xjc.api.impl.s2j.SchemaCompilerImpl.bind(SchemaCompilerImpl.java:269)
[INFO] 	at com.sun.tools.xjc.api.impl.s2j.SchemaCompilerImpl.bind(SchemaCompilerImpl.java:95)
[INFO] 	at com.sun.tools.ws.processor.modeler.wsdl.JAXBModelBuilder.bind(JAXBModelBuilder.java:142)
[INFO] 	at com.sun.tools.ws.processor.modeler.wsdl.WSDLModeler.buildJAXBModel(WSDLModeler.java:2298)
[INFO] 	at com.sun.tools.ws.processor.modeler.wsdl.WSDLModeler.internalBuildModel(WSDLModeler.java:198)
[INFO] 	at com.sun.tools.ws.processor.modeler.wsdl.WSDLModeler.buildModel(WSDLModeler.java:141)
[INFO] 	at com.sun.tools.ws.wscompile.WsimportTool.buildWsdlModel(WsimportTool.java:444)
[INFO] 	at com.sun.tools.ws.wscompile.WsimportTool.run(WsimportTool.java:205)
[INFO] 	at com.sun.tools.ws.wscompile.WsimportTool.run(WsimportTool.java:183)
[INFO] 	... 5 more
[INFO] Caused by: org.xml.sax.SAXParseException; systemId: jar:file:/P:/git/jaxws-maven-plugin/target/it-repo/com/sun/xml/bind/jaxb-xjc/2.2.8-b01/jaxb-xjc-2.2.8-b01.jar!/com/sun/tools/xjc/reader/xmlschema/bindinfo/binding.xsd; lineNumber: 52; columnNumber: 88; schema_reference: Schemadokument 'xjc.xsd' konnte nicht gelesen werden, weil der 'file'-Zugriff wegen der von der Eigenschaft accessExternalSchema festgelegten Einschr寫ung nicht zul崳ig ist. 
[INFO] 	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(ErrorHandlerWrapper.java:203)
[INFO] 	at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(ErrorHandlerWrapper.java:177)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(XMLErrorReporter.java:400)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.reportSchemaErr(XSDHandler.java:4156)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.reportSchemaFatalError(XSDHandler.java:4135)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.getSchemaDocument(XSDHandler.java:2172)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.resolveSchema(XSDHandler.java:2082)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.constructTrees(XSDHandler.java:1012)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.parseSchema(XSDHandler.java:623)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadSchema(XMLSchemaLoader.java:613)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadGrammar(XMLSchemaLoader.java:572)
[INFO] 	at com.sun.org.apache.xerces.internal.impl.xs.XMLSchemaLoader.loadGrammar(XMLSchemaLoader.java:538)
[INFO] 	at com.sun.org.apache.xerces.internal.jaxp.validation.XMLSchemaFactory.newSchema(XMLSchemaFactory.java:255)
[INFO] 	at javax.xml.validation.SchemaFactory.newSchema(SchemaFactory.java:638)
[INFO] 	at javax.xml.validation.SchemaFactory.newSchema(SchemaFactory.java:670)
[INFO] 	at com.sun.tools.xjc.SchemaCache.newValidator(SchemaCache.java:77)
[INFO] 	... 16 more
[INFO] [INFO] ------------------------------------------------------------------------
[INFO] [INFO] BUILD FAILURE
[INFO] [INFO] ------------------------------------------------------------------------
```

**Comment by phax** [comment]
I re-test with your updates on #8

**Comment by phax** [comment]
Now I have
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```
so fine for me :)

**Comment by andham** [comment]
The status on the jaxwscommons-43 IT is that the specified mavenOpts for the test (specified in invoker.properties) isn't applied when executing on Travis. This caused the IT to fail.
However, locally on Windows, Linux, and MacOS it works.

**Comment by andham** [comment]
I'll re-write the jaxwscommons-43 IT to work around this. This will actual improve the test.

**Comment by andham** [comment]
Rewrite of jaxwscommons-43 done in 468e202.
Now only openjdk7 fails with jaxwscommons-81.

**Comment by andham** [comment]
For the IT jaxwscommons-81 on openjdk7 I see the following in the build log:
> [INFO] (process:4274): GLib-GIO-ERROR **: Settings schema 'org.gnome.system.proxy' is not installed

Installing ubuntu package 'gsettings-desktop-schemas' solves this. Fixed in 5a4617f.

Now all ITs build successfully on Travis CI. Let's hope it stays that way...
